providers/structured/provider.go

```diff
@@ -47,16 +50,25 @@ func (p StructuredOutputLogProvider) GetRecordEventCalls() []RecordEventCallArgs
 
 // NewStructuredOutputLogProvider returns a LogProvider which records all calls made to it.
 func NewStructuredOutputLogProvider() *StructuredOutputLogProvider {
+	return LogProvider(nil).(*StructuredOutputLogProvider)
+}
+
+func LogProvider(nextProvider providers.LogProvider) providers.LogProvider {
```

Comment:
Why do we need two different constructors here, with different signatures?

Reply:
Because `NewStructuredOutputLogProvider()` is already being used in other services. I didn't wanna break any existing functionality.

Reply:
Hmm, OK. Let's mark the old one deprecated using the Go convention (see: https://rakyll.org/deprecated/), and have this also return *StructuredOutputLogProvider to avoid awkward casting of its result.